lib/BackbeatProducer.js

```diff
@@ -128,17 +128,29 @@ class BackbeatProducer extends EventEmitter {
                 cb(errors.InternalError);
             });
         }
-        const payload = this._partition === undefined ?
-                  entries.map(item => ({
-                      topic: this._topic,
-                      messages: new KeyedMessage(item.key, item.message),
-                      key: item.key,
-                  })) :
-                  entries.map(item => ({
-                      topic: this._topic,
-                      messages: item.message,
-                      partition: this._partition,
-                  }));
+
+        const payload = entries.map(item => {
+            if (item.partition) {
```

Comment:
This needs to be `if (item.partition !== undefined)` since partition can be `0`.